--- Src/Zle/zle_hist.c
+++ Src/Zle/zle_hist.c
@@ -558,7 +558,7 @@
 
 /**/
 int
-getline(char **args)
+zlegetline(char **args)
 {
     char *s = (char *)getlinknode(bufstack);
 
--- Src/Zle/iwidgets.list
+++ Src/Zle/iwidgets.list
@@ -56,7 +56,7 @@
 "expand-word", expandword, 0
 "forward-char", forwardchar, 0
 "forward-word", forwardword, 0
-"get-line", getline, 0
+"get-line", zlegetline, 0
 "gosmacs-transpose-chars", gosmacstransposechars, 0
 "history-beginning-search-backward", historybeginningsearchbackward, 0
 "history-beginning-search-forward", historybeginningsearchforward, 0
